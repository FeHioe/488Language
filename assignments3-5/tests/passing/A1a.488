% A1 Part A: a program that uses all arithmetic, logical and comparison operators
% Author: Ishtiaque Khaled - CDF: khaledi1


% Main program scope
% Code was intentionally written with different styles to add more variability for the compiler testcases
{
	var a : Integer
	var c : Boolean

	% Arithmetic operations
	% Written with various whitespaces between operators and variables
	a : = 1
	a := -1
	write a, " ArithExpn NEG - Expected a = -1", newline

	a := 1+1
	write a, " ArithExpn ADD - Expected a = 2", newline

	a : = 1-1
	write a, " ArithExpn SUB - Expected a = 0", newline

	a : = 1 * 1
	write a, " ArithExpn MUL - Expected a = 1", newline

	a := 1 / 1
	write a, " ArithExpn DIV - Expected a = 1", newline

	a := 1 + 1 - 1 * 1 / 1
  write a, " ArithExpn Combination - Expected a = 1", newline
	% Logical and comparison operators

	c : = true

	c : = false

	c : = not true

	c : = true and false

	c : = true or false

	c : = 1 = 1

	c : = 1 not = 1

	c : = 1 < 2

	c : = 1 > 2

	c : = 1 >= 1

	c : = 1 <= 1

}
