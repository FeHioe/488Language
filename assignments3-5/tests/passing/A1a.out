-1 ArithExpn NEG - Expected a = -1
2 ArithExpn ADD - Expected a = 2
0 ArithExpn SUB - Expected a = 0
1 ArithExpn MUL - Expected a = 1
1 ArithExpn DIV - Expected a = 1
1 ArithExpn Combination - Expected a = 1

